### PR TITLE
Port D1 loadout builder to React

### PR DESCRIPTION
**Pull request by bhollis**
This moves the old D1 loadout builder to React, and cleans it up just a bit in the process. Only very minor graphical changes (such as using the standard character selector and collapsible area). Otherwise everything works.

As this is one of the last major AngularJS pieces, I was able to drop a bunch of big dependencies like ngDialog and angular drag-drop.

screen shot 2019-02-05 at 1 12 33 am

### CODE REVIEW

**Comment by kyleshay** [comment]
Daaaaang :O This is awesome!! Haha, I actually started looking at this last night. Major props, huge chunk gone :D Deleting those old angular files has got to be satisfying!

**Comment by delphiactual** [comment]
`Uncaught TypeError: Cannot read property 'defaultCollapsed' of undefined`

**Comment by delphiactual** [comment]
Can also remove `ng-i18next` and `angular-sanitize`

File: src/app/dim-ui/CollapsibleTitle.tsx [hunk 1]
```diff
@@ -25,8 +26,9 @@ interface DispatchProps {
 }
 
 function mapStateToProps(state: RootState, props: ProvidedProps): StoreProps {
+  const collapsed = state.settings.collapsedSections[props.sectionId];
   return {
-    collapsed: state.settings.collapsedSections[props.sectionId]
+    collapsed: collapsed === undefined ? !this.props.defaultCollapsed : collapsed
```
Comment (delphiactual) [inline src/app/dim-ui/CollapsibleTitle.tsx]: `collapsed: collapsed === undefined ? !props.defaultCollapsed : collapsed`

**Comment by bhollis** [comment]
ng-i18next is still used by item reviews.

File: src/app/dim-ui/CollapsibleTitle.tsx (hunk 1 above)
Comment (bhollis) [reply src/app/dim-ui/CollapsibleTitle.tsx]: 👍

**Comment by delphiactual** [comment]
`ng-i18next` depends on `angular-sanitize`

**Comment by bhollis** [comment]
Fantastic. I'll add it back in then.

**Pull request merged**